examples/player/simid_player.js

```diff
@@ -609,16 +609,69 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when video playback is complete.
+   * @private
+   */
+  videoComplete() {
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
+
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+    }
+
+    //If the request duration is longer than the ad duration, the ad extends for the requested amount of time
+    if (this.requestedDuration_ != UNLIMITED_DURATION) {
+      const durationChangeMs = (this.requestedDuration_ - this.adVideoElement_.duration) * 1000;
+      setTimeout(() => {
+        this.stopAd(StopCode.CREATIVE_INITIATED);
+      }, durationChangeMs);
+    }
+  }
+
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    const originalDuration = this.requestedDuration_;
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    if (this.requestedDuration_ != UNLIMITED_DURATION && this.requestedDuration_< 0) {
+      const durationErrorMessage = {
+        errorCode: PlayerErrorCode.UNSUPPORTED_TIME,
+        message: 'A negative duration is not valid.'
+      }
+      this.requestedDuration_ = originalDuration;
+      this.simidProtocol.reject(incomingMessage, durationErrorMessage);
+    }
+    else {
+      //If requested duration is any other acceptable value
+      this.compareAdAndRequestedDurations();
+      this.simidProtocol.resolve(incomingMessage);
     }
-    const requestedDuration  = incomingMessage.args['duration'];
-    this.requestedDuration_ = requestedDuration;
-    this.simidProtocol.resolve(incomingMessage);
   }
 
+  /**
+   * Compares the duration of the ad with the requested change duration.
+   * If request duration is the same as the ad duration, ad ends as normal.
+   * If request duration is unlimited, ad stays on screen until user closes ad.
+   * If request duration is shorter, the ad stops early. 
+   * @private
+   */
+  compareAdAndRequestedDurations() {
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION || 
+      this.requestedDuration_ == UNLIMITED_DURATION) {
+        //Note: Users can end the ad with unlimited duration with
+        // the close ad button on the player
+        return;
+      }
+    else if (this.adVideoElement_.currentTime >= this.requestedDuration_) {
+      //Creative requested a duration shorter than the ad
+      this.stopAd(StopCode.CREATIVE_INITATED);
+      return;
```

Comment:
You don't need both the returns and the else/if for this function, either one or the other should suffice

Reply:
deleted return